app/models/katello/system.rb

```diff
@@ -23,8 +23,8 @@ class System < Katello::Model
     has_many :system_activation_keys, :class_name => "Katello::SystemActivationKey", :dependent => :destroy
     has_many :activation_keys, :through => :system_activation_keys
 
-    has_many :system_host_collections, :class_name => "Katello::SystemHostCollection", :dependent => :destroy
-    has_many :host_collections, :through      => :system_host_collections
+    has_many :host_collection_hosts, :class_name => "Katello::HostCollectionHosts", :dependent => :destroy, :foreign_key => :host_id
+    has_many :host_collections, :through      => :host_collection_hosts
```

Comment:
You can't create the association to :host_collection_hosts because :host_id isn't the system id.

You can create a has_many :host_collections :through => :foreman_host  I *think*